### DISCUSSION TITLE
Question - Using Proxies

### DISCUSSION THREAD

**Discussion by EdgeSync** [CrowdStrike/falconpy#154] [Q&A] [answered]
Hi team, 

I am not able to find anything in the documentation which details assigning a proxy for FalconPy functionality. 

Is there some way to assign a proxy, or do we need to use some other method? 

    def falcon_auth(self):
        authorization = FalconAuth.OAuth2(creds={
            "client_id": self.client_id,
            "client_secret": self.client_secret,
            },
            base_url=self.base_url)
            self.token = authorization.token()['"body"]["access_token"]

**Comment by jshcodes** [comment]
Hi @EdgeSync -

Currently FalconPy is unaware of the existing proxy configuration, and should use whatever proxy is configured for regular TCP or HTTPS traffic from the machine executing the code.

> _Please note_: FalconPy does have functionality for disabling SSL Verification should you need to offload SSL, have a legacy pinning solution in place, etc.

**Comment by EdgeSync** [comment, marked as answer]
hey @jshcodes, thanks for the response. 

Okay sure - thanks for the input. I wanted to make 100% sure it wasn't a feature somewhere that I missed. I think we can all be guilty of looking at documentation for too long and miss the obvious :laughing: 

In case anyone also requires proxy configuration, you can get around the lack of proxy support in a number of ways, either by wrapping the PyFalcon function calls in an additional proxy library, or by setting the proxy using os.environ["http_proxy"] / os.environ["https_proxy"]  variables, or having them already set in your environment variables before running the script.
